src/amqp.rs

```diff
@@ -0,0 +1,95 @@
+use lapin::tcp::{OwnedIdentity, OwnedTLSConfig};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use tokio_amqp::*;
+
+/// Client certificate for rabbit authentication
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub(crate) enum ClientCertDer {
+    /// Certificate embedded in config file as base64 string
+    Embedded(String),
+    /// Certificate on file system
+    Path(PathBuf),
+}
+
+/// Tls identity configuration parameters
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub(crate) struct TlsIdentity {
+    /// Client certificate and key
+    pub client_cert_and_key_der: ClientCertDer,
+    /// Password for client cert and key
+    #[serde(default)]
+    pub client_cert_and_key_password: String,
+}
+
+/// Tls configuration parameters
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub(crate) struct AmpqTlsConfig {
+    /// Client certificate
+    pub identity: Option<TlsIdentity>,
+    /// CA certificate to use to establish trust
+    pub ca_cert: PathBuf,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct AmqpConfig {
+    /// Format: amqp://user:password@host:port/vhost?timeout=seconds
+    pub(crate) endpoint: String,
+    pub(crate) tls: Option<AmpqTlsConfig>,
```

Comment:
I like that you added TLS configuration here! Do you think it'd be possible to bring this configuration in-line with the TLS configuration we have on other components so that it supports the same options? That is, replace this `AmqpTlsConfig` struct with the one defined here:

https://github.com/timberio/vector/blob/410fac0b7fbb0c7361056105ba362f7ee6c112ca/src/tls/settings.rs#L28-L32